src/Command/RegisterModuleCommand.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\Mvc\Command;
+
+use Laminas\Cli\CommandListenerInterface;
+use Laminas\ComponentInstaller\Collection;
+use Laminas\ComponentInstaller\ConfigDiscovery;
+use Laminas\ComponentInstaller\ConfigOption;
+use Laminas\ComponentInstaller\Injector\DevelopmentConfigInjector;
+use Laminas\ComponentInstaller\Injector\DevelopmentWorkConfigInjector;
+use Laminas\ComponentInstaller\Injector\InjectorInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Event\ConsoleEvent;
+use Symfony\Component\Console\Helper\QuestionHelper;
+use Symfony\Component\Console\Input\ArrayInput;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\ConfirmationQuestion;
+
+final class RegisterModuleCommand extends Command implements CommandListenerInterface
+{
+    /** @var string */
+    protected static $defaultName = 'mvc:module:register';
+
+    protected function configure(): void
+    {
+        $this->setName(self::$defaultName);
+        $this->setDescription('Register given module within the application');
+        $this->addOption(
+            'module',
+            null,
+            InputOption::VALUE_OPTIONAL,
+            'Module name to register'
+        );
+        $this->addOption(
+            'dir',
+            null,
+            InputOption::VALUE_OPTIONAL,
+            'Directory with modules'
+        );
+        $this->addOption(
+            'mode',
+            null,
+            InputOption::VALUE_OPTIONAL,
+            'Production or development mode'
```

Comment:
Same comment here as I had for the `EnableComposerAutoloadingCommand`. Maybe these should be constants defined in a marker interface?